### DISCUSSION TITLE
Complex firing solutions to include multiple rounds landing simultaneously

### DISCUSSION THREAD

**Discussion by alexandercarruthers** [sh4rkman/SquadCalc#158] [Ideas]
Is it possible in Squad to have multiple mortar rounds land at the same time by varying the mortar angle alone?

A very quick sketch up in Python tells me that I don't think it is possible due to the inability to control the initial muzzle velocity of the mortars. 

Here is some code that brought me to my initial findings but keen to hear if you have a different or alternative opinion?

I have not included the time to reload the mortar so this is very much an academic exercise for the moment.

The graph below notes that the 2 mortar rounds would impact where they intersect which is 300m away from the launch point at a height of 250m (ish). 

![image](https://github.com/sh4rkman/SquadCalc/assets/48217309/367e9083-e54f-4ce4-896a-e17dcf512cf5)

```python
import numpy as np
import matplotlib.pyplot as plt

# Constants
g = 9.81  # acceleration due to gravity in m/s^2

# Given values
theta1_mrad = 800  # given angle in milliradians
theta2_mrad = 1400
v = 109.890938  # initial velocity in m/s

# Convert angle to radians
theta1_rad = theta1_mrad / 1000
theta2_rad = theta2_mrad / 1000

T1 = (2 * v * np.sin(theta1_rad)) / g
T2 = (2 * v * np.sin(theta2_rad)) / g

# Time intervals
t = np.linspace(0, T2, num=1000)

# Trajectory for theta1
x1 = v * np.cos(theta1_rad) * t
y1 = v * np.sin(theta1_rad) * t - 0.5 * g * t**2

x2 = v * np.cos(theta2_rad) * t
y2 = v * np.sin(theta2_rad) * t - 0.5 * g * t**2

# Plotting the trajectories
plt.figure(figsize=(10, 6))
plt.plot(x1, y1, label=f'Theta1 = {theta1_mrad:.2f} milliradians')
plt.plot(x2, y2, label=f'Theta2 = {theta2_mrad:.2f} milliradians')
plt.xlabel('Horizontal Distance (m)')
plt.ylabel('Vertical Distance (m)')
plt.title('Trajectories at Different Angles to Land at the Same Distance')
plt.legend()
plt.grid(True)
plt.show()
```

**Comment by sh4rkman** [comment]
`Is it possible in Squad to have multiple mortar rounds land at the same time by varying the mortar angle alone?`

At the same time **and the same target ?**

**Reply by alexandercarruthers** [reply]
Same target and same time, so 2 rounds from 1 mortar tube land simultaneously.

This article will describe it better than I can https://en.wikipedia.org/wiki/Artillery#Multiple_round_simultaneous_impact

**Reply by sh4rkman** [reply]
Theorically, the formula that resolve the angle needed to reach a target has two solutions (https://en.wikipedia.org/wiki/Projectile_motion#Angle_%CE%B8_required_to_hit_coordinate_(x,_y)) : 

1. One "Low Angle" solution : 0° < solution1 < 45°
2. One "High Angle" solution : 45° < solution2 < 90°

Thing is that in Squad, most mortars are locked ingame in 0-45 or 45-90, therefore they have one single angle solution for hitting a target. Some weapons (BM-21 GRAD comes to my mind) can actually go from low to high angle and have two different solution available for them.

The low angles trajectory are obviously faster, so in these case it's theorically possible to shoot High angle first and low angle then.

You an use squadcalc legacy mode to swith from high to low angle to play around : 

![20240616_17h04m_inLCUGKzaVf](https://github.com/sh4rkman/SquadCalc/assets/9033912/296b3b7a-bfa0-480f-bad9-efc0777a25be)

**Comment by sh4rkman** [comment]
You can now switch between high an low angle by clicking any weapon : 

![image](https://github.com/user-attachments/assets/26b3a94f-8944-4338-a7f0-1a0898756d84)

That means you now have 2 firing solutions with different travel time, from there you can shoot the first firing solution (high one), count the travel time difference then shoot the second one (low one).

Wrote a bit in the new wiki about it and the maths behind it : https://github.com/sh4rkman/SquadCalc/wiki/Deducing-Elevation#high--low-angles
